mantidproject/mantid
# Add option to highlight zero detectors in Instrument Viewer

**Description of work.**

Adds new checkbox to allow detectors with zero counts to be highlighted. These aren't distinguished very clearly from detectors with low counts on the color scale. See linked issue for more details

Notes: 
The function InstrumentWidgetRenderTab::changeColorMap isn't currently exposed to python in qt/python/mantidqt/mantidqt/widgets/instrumentview/api.py therefore this didn't need changing
I've checked that project save, load still works on old .mtdproj files that don't contain this new property

**To test:**

Load run SXD33304 and open up instrument viewer
Open up the Color Map dialog from the Display Settings drop down on the Render tab
Check the new "Highlight Detectors With Zero Counts" checkbox
Note the instrument viewer now shows a lot of detectors greyed out

Fixes #34177.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
